src/context.rs

```diff
@@ -668,6 +756,92 @@ mod tests {
         }
     }
 
+    mod before_all {
```

Comment:
(tested) properties of the `before_all`:
- [x] can be called inside describe
- [x] can be called inside nested describe
- [ ] it will be called before any `it` of its group
- [ ] it will be called only once

I understand it's the same test over and over again. This is a smell. If we break a property of `before_all`, only one test should break. We should do better.